test/test_visualization.jl

```diff
@@ -19,7 +20,6 @@ isdir(outdir) && rm(outdir, recursive=true)
   test_examples_2d = Dict(
     "TreeMesh" => ("tree_2d_dgsem", "elixir_euler_blast_wave_amr.jl"),
     "StructuredMesh" => ("structured_2d_dgsem", "elixir_euler_source_terms_waving_flag.jl"),
-    "UnstructuredMesh2D" => ("unstructured_2d_dgsem", "elixir_euler_wall_bc.jl")
```

Comment:
Why did you remove this one?

Reply:
This PR causes the old recipes to fail because we're pirating the `PlotData2D(...)` API. However, the old recipes for `UnstructuredMesh2D` were just scatter plots, and were essentially a placeholder until more mature functionality was introduced. 

I'm not sure of the best way around this conflict. We break Plots.jl functionality for `UnstructuredMesh2D` in this PR but won't replace it until #755 or #785. Perhaps we could merge this PR to `dev`, then merge #755 and consolidate them inside `dev`?

Reply:
If UnstructuredMesh2D really only used scatter plots, I don't think we lose anything by breaking support for it. Especially since we have support for proper plotting coming in from @FelipeSantillan's PR. Thus unless @andrewwinters5000 objects, I'd say out the door with it!

Reply:
I was worried previously that there was some slice support in `UnstructuredMesh2D` that I was erasing, but it looks like that's not the case. In other words, this is a direct replacement of plotting capabilities.

Reply:
Okay, let's go ahead :+1: